packages/libs/client/src/composePactCommand/composePactCommand.ts

```diff
@@ -63,62 +20,58 @@ interface IComposePactCommand {
         | ((payload: TPayload & Partial<IPactCommand>) => Partial<IPactCommand>)
       >,
     ]
-  ): (cmd?: Partial<IPactCommand>) => Partial<IPactCommand>;
+  ): (cmd?: InitialInput) => Partial<IPactCommand>;
 
   (
     first: DataOrFunction<NoPayload<Partial<IPactCommand>>>,
     ...rest: Array<DataOrFunction<Partial<IPactCommand>>>
-  ): (cmd?: Partial<IPactCommand>) => Partial<IPactCommand>;
+  ): (cmd?: InitialInput) => Partial<IPactCommand>;
 }
 
+const finalizeCommand = (
+  command: Partial<IPactCommand>,
+): Partial<IPactCommand> => {
+  const dateInMs = Date.now();
+  const finalCommand = { ...command };
+  finalCommand.nonce ??= `kjs:nonce:${dateInMs}`;
+  finalCommand.signers ??= [];
```

Comment:
Did you check whether this is supported properly? Or is it compiled away by `tsc`?

Reply:
it compiled to this by tsc
```js
 const finalCommand = Object.assign({}, command);
 (_a = finalCommand.nonce) !== null && _a !== void 0 ? _a : (finalCommand.nonce = `kjs:nonce:${dateInMs}`);
 (_b = finalCommand.signers) !== null && _b !== void 0 ? _b : (finalCommand.signers = []);
```